### ISSUE TITLE
npm install fails with 404 for atom.io

### ISSUE THREAD

**Issue by Jeansen** [tuxedocomputers/tuxedo-control-center#250]
Clone the repo and followed the README. Doing 'npm install' results in  404 response downloading https://atom.io/download/electron/v13.6.9/node-v13.6.9-headers.tar.gz

```
 tuxedo-control-center@1.2.2 postinstall /mnt/data/git-clones/tuxedo-control-center
> electron-builder install-app-deps

  • electron-builder version=20.44.4
  • rebuilding native production dependencies platform=linux arch=x64
Error: /home/marcel/.config/nvm/versions/node/v14.21.2/bin/node exited with code 1
Output:

> abstract-socket@2.1.1 install /mnt/data/git-clones/tuxedo-control-center/node_modules/abstract-socket
> node-gyp rebuild


Error output:
gyp WARN install got an error, rolling back install
gyp ERR! configure error 
gyp ERR! stack Error: 404 response downloading https://atom.io/download/electron/v13.6.9/node-v13.6.9-headers.tar.gz
gyp ERR! stack     at Request.<anonymous> (/home/marcel/.config/nvm/versions/node/v14.21.2/lib/node_modules/npm/node_modules/node-gyp/lib/install.js:188:14)
gyp ERR! stack     at Request.emit (events.js:412:35)
gyp ERR! stack     at Request.onRequestResponse (/home/marcel/.config/nvm/versions/node/v14.21.2/lib/node_modules/npm/node_modules/request/request.js:1066:10)
gyp ERR! stack     at ClientRequest.emit (events.js:400:28)
gyp ERR! stack     at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:647:27)
gyp ERR! stack     at HTTPParser.parserOnHeadersComplete (_http_common.js:127:17)
gyp ERR! stack     at TLSSocket.socketOnData (_http_client.js:515:22)
gyp ERR! stack     at TLSSocket.emit (events.js:400:28)
gyp ERR! stack     at addChunk (internal/streams/readable.js:293:12)
gyp ERR! stack     at readableAddChunk (internal/streams/readable.js:267:9)
gyp ERR! System Linux 6.0.0-6-amd64
gyp ERR! command "/home/marcel/.config/nvm/versions/node/v14.21.2/bin/node" "/home/marcel/.config/nvm/versions/node/v14.21.2/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /mnt/data/git-clones/tuxedo-control-center/node_modules/abstract-socket
gyp ERR! node -v v14.21.2
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! abstract-socket@2.1.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the abstract-socket@2.1.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/marcel/.npm/_cacache/_logs/2022-12-16T22_38_45_581Z-debug.log

    at ChildProcess.<anonymous> (/mnt/data/git-clones/tuxedo-control-center/node_modules/builder-util/src/util.ts:244:14)
    at Object.onceWrapper (events.js:520:26)
    at ChildProcess.emit (events.js:400:28)
    at maybeClose (internal/child_process.js:1088:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:296:5)
From previous event:
    at _rebuild (/mnt/data/git-clones/tuxedo-control-center/node_modules/app-builder-lib/src/util/yarn.ts:179:5)
    at rebuild (/mnt/data/git-clones/tuxedo-control-center/node_modules/app-builder-lib/out/util/yarn.js:163:19)
    at /mnt/data/git-clones/tuxedo-control-center/node_modules/app-builder-lib/src/util/yarn.ts:20:11
From previous event:
    at _installOrRebuild (/mnt/data/git-clones/tuxedo-control-center/node_modules/app-builder-lib/out/util/yarn.js:71:28)
    at installOrRebuild (/mnt/data/git-clones/tuxedo-control-center/node_modules/app-builder-lib/out/util/yarn.js:55:28)
    at /mnt/data/git-clones/tuxedo-control-center/node_modules/electron-builder/src/cli/install-app-deps.ts:58:9
    at Generator.next (<anonymous>)
    at processImmediate (internal/timers.js:464:21)
From previous event:
    at _installAppDeps (/mnt/data/git-clones/tuxedo-control-center/node_modules/electron-builder/out/cli/install-app-deps.js:177:26)
    at installAppDeps (/mnt/data/git-clones/tuxedo-control-center/node_modules/electron-builder/out/cli/install-app-deps.js:146:26)
    at /mnt/data/git-clones/tuxedo-control-center/node_modules/electron-builder/src/cli/cli.ts:49:19
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! tuxedo-control-center@1.2.2 postinstall: `electron-builder install-app-deps`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the tuxedo-control-center@1.2.2 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

```

**Comment by LonelyMoose** [comment]
Had the same problem, apparently atom [deprecated their website](https://github.blog/2022-06-08-sunsetting-atom/). You can temporarily fix that by downloading headers from their [new website](https://artifacts.electronjs.org/headers/dist/v13.6.9/node-v13.6.9-headers.tar.gz) (as specified [here](https://www.electronjs.org/blog/s3-bucket-change)), and load them using `npm config set tarball <path-to-headers>` and then running `npm install`.

**Comment by misterhsp** [comment]
I have the same problem.

...

**Comment by Jeansen** [comment]
@LonelyMoose Thank your for the detailed information.  That worked like a charm! Unfortunately even in the latest version v1.2.3 this issue still persists.

**Comment by srogmann** [comment]
atom isn't supported any longer.

Quote from https://github.blog/2022-06-08-sunsetting-atom/:

> If I’m using Atom, what changes can I expect after the sunset?
>
> No more security updates

**Comment by misterhsp** [comment]
When is this going to get fixed? I am at the build of 1.2.5 again fully run in.

...

**Comment by Kangie** [comment]
Fixed for 2.0.x
